crates/egui-wgpu/src/winit.rs

```diff
@@ -235,20 +203,20 @@ impl Painter {
     ///
     /// # Errors
     /// If the provided wgpu configuration does not match an available device.
-    pub async unsafe fn set_window(
+    pub async fn set_window(
         &mut self,
         window: Option<&winit::window::Window>,
     ) -> Result<(), crate::WgpuError> {
         match window {
             Some(window) => {
-                let surface = self.instance.create_surface(&window)?;
+                let surface = unsafe { self.instance.create_surface(&window)? };
 
                 self.ensure_render_state_for_surface(&surface).await?;
+                let render_state = self.render_state.as_ref().unwrap();
```

Comment:
`unwrap` 😱  - please return an error instead, or restructure the code so it is impossible to hit

Reply:
Hmm I tried to return a reference from the line above, but can't do that because it takes ownership of `self`. Probably that method can just be inlined here and done